core/modules/layout/layout.admin.inc

```diff
@@ -326,28 +348,71 @@ function layout_settings_form($form, &$form_state, Layout $layout) {
   $form['context_wrapper']['context']['required'] = array(
     '#theme' => 'layout_settings_context_table',
     '#layout_path' => $layout->getPath(),
+    '#access' => count($contexts) > $built_in_contexts,
   );
 
   foreach ($contexts as $context_key => $layout_context) {
-    if ($layout_context->position && $layout_context->locked) {
-      $form['context_wrapper']['context']['required'][$context_key]['plugin'] = array(
-        '#type' => 'value',
-        '#value' => $layout_context->plugin,
-      );
-      $form['context_wrapper']['context']['required'][$context_key]['label'] = array(
-        '#markup' => check_plain($all_context_info[$layout_context->plugin]['title']),
-      );
-    }
-    elseif ($layout_context->position && $layout_context->required) {
-      $form['context_wrapper']['context']['required'][$context_key]['plugin'] = array(
-        '#type' => 'select',
-        '#options' => $context_options,
-        '#default_value' => $layout_context->plugin,
-      );
-    }
-    else {
-      // @todo: Support adding and removing custom contexts not based on path.
-      // e.g. Adding a Node context based on a hard-coded ID.
+    if (!in_array($context_key, array('current_user', 'overrides_path', 'dashboard'))) {
```

Comment:
Is there a way we can avoid hard-coding these 3 exceptions?

Reply:
Would still like if we could avoid these exceptions. Perhaps either a method or special key on the context, e.g. `$layout_context->isConfigurable()`.